plugins/jobtype/test/azkaban/jobtype/TestHadoopSecureSparkWrapper.java

```diff
@@ -0,0 +1,143 @@
+    setEnv(envs);
+    Configuration.addDefaultResource("yarn-site.xml");
+    String[] argArray = new String[] {
+        "--conf",
+        "spark.yarn.queue=test",
+        "--conf",
+        "spark.yarn.executor.nodeLabelExpression=test",
+        "--executor-cores",
+        "2",
+        "--executor-memory",
+        "7G"
+    };
+    argArray = HadoopSecureSparkWrapper.handleNodeLabeling(argArray);
+    argArray = HadoopSecureSparkWrapper.removeNullsFromArgArray(argArray);
+    Assert.assertTrue(argArray.length == 6);
+    Assert.assertTrue(argArray[1].contains("test2"));
+  }
+
+  @Test
+  public void testDisableAutoLabeling() {
+    Map<String, String> envs = new TreeMap<String, String>(String.CASE_INSENSITIVE_ORDER);
+    envs.put(HadoopSparkJob.SPARK_NODE_LABELING_ENV_VAR, Boolean.TRUE.toString());
+    envs.put(HadoopSparkJob.SPARK_DESIRED_NODE_LABEL_ENV_VAR, "test2");
+    envs.put(HadoopSparkJob.SPARK_MIN_MEM_VCORE_RATIO_ENV_VAR, "3");
+    URL url = getClass().getClassLoader().getResource("spark-defaults.conf");
+    envs.put(HadoopSparkJob.SPARK_PROPERTY_FILE_PATH_ENV_VAR, url.getPath());
+    setEnv(envs);
+    Configuration.addDefaultResource("yarn-site.xml");
+    String[] argArray = new String[] {
+        "--conf",
+        "spark.yarn.queue=test",
+        "--conf",
+        "spark.yarn.executor.nodeLabelExpression=test",
+        "--executor-cores",
+        "2",
+        "--executor-memory",
+        "7G"
+    };
+    argArray = HadoopSecureSparkWrapper.handleNodeLabeling(argArray);
+    argArray = HadoopSecureSparkWrapper.removeNullsFromArgArray(argArray);
+    Assert.assertTrue(argArray.length == 6);
+    Assert.assertTrue(argArray[1].contains("test"));
+  }
+
+  @Test
+  public void testLoadConfigFromPropertyFile() {
+    Map<String, String> envs = new TreeMap<String, String>(String.CASE_INSENSITIVE_ORDER);
+    envs.put(HadoopSparkJob.SPARK_NODE_LABELING_ENV_VAR, Boolean.TRUE.toString());
+    envs.put(HadoopSparkJob.SPARK_AUTO_NODE_LABELING_ENV_VAR, Boolean.TRUE.toString());
+    envs.put(HadoopSparkJob.SPARK_DESIRED_NODE_LABEL_ENV_VAR, "test2");
+    envs.put(HadoopSparkJob.SPARK_MIN_MEM_VCORE_RATIO_ENV_VAR, "3");
+    URL url = getClass().getClassLoader().getResource("spark-defaults.conf");
+    envs.put(HadoopSparkJob.SPARK_PROPERTY_FILE_PATH_ENV_VAR, url.getPath());
+    setEnv(envs);
+    Configuration.addDefaultResource("yarn-site.xml");
+    String[] argArray = new String[] {
+        "--conf",
+        "spark.yarn.queue=test",
+        "--conf",
+        "spark.yarn.executor.nodeLabelExpression=test"
+    };
+    argArray = HadoopSecureSparkWrapper.handleNodeLabeling(argArray);
+    argArray = HadoopSecureSparkWrapper.removeNullsFromArgArray(argArray);
+    Assert.assertTrue(argArray.length == 2);
+    Assert.assertTrue(argArray[1].contains("test2"));
+  }
+
+  @SuppressWarnings("unchecked")
+  private static void setEnv(Map<String, String> newenv) {
```

Comment:
I don't think so.
Since the code fetches these parameters through env var, there're very limited option in terms of manipulating the env var to control the parameters passed to the actual code.